lib/tasks/index.rake

```diff
@@ -0,0 +1,54 @@
+require 'lockfile'
+
+INDEX_DIR = Pathname.new('/web-archiving-stacks/data/indices/cdx').freeze
+
+namespace :index do
+  desc "Show indexes"
+  task :show do
+    system("ls -l #{INDEX_DIR}/level?.cdx")
+  end
+
+  namespace :rollup do
+    def rollup(n) # n is the log level from which to rollup
+      src = INDEX_DIR.join("level#{n}.cdx")
+      dst = INDEX_DIR.join("level#{n + 1}.cdx")
+
+      Lockfile.new(INDEX_DIR.join('working.lock').to_s) do # ensure only 1 rollup process
+        raise "Missing #{src} file" unless src.exist?
+        raise "Missing #{dst} file" unless dst.exist?
+
+        return if src.zero? # nothing to do
+
+        oflags = '--buffer-size=128M' # merge optimization flags
+        ENV['LC_ALL'] = 'C'.freeze # native sort order
+        ENV['TMPDIR'] = '/web-archiving-stacks/data/tmp'.freeze # use big tmpdir
+        # we need to merge the 2 *already sorted* files into the new file, and
+        # then zero out the current level of index
+        if system("sort --merge --unique #{oflags} --output=#{dst}.new #{src} #{dst}")
```

Comment:
will we have a problem if the index is actively being updated while this is going on?  (most crucial for level0 to level1, but if a rake task doesn't finish and we don't notice ..)